site/docs/snapshot-tag-branch.md

```diff
@@ -0,0 +1,148 @@
+    In a table, the lack of a file conflict does not mean that the transaction can be committed.
+    In addition, longer transaction lengths from branch-like behavior dramatically increases the likelihood that the transaction could fail.
+    The merge feature would likely be supported through multi-table transaction in the future.
+
+## Snapshot Reference
+
+In version control systems like git, branch and tag are both references of commits.
+In Iceberg, we use a similar concept of **Snapshot Reference** to implement branching and tagging.
+
+Each Iceberg table metadata contains a list of `refs` (references), and a `current-branch` indicating the current branch to use.
+When user creates an Iceberg table, the first commit belongs to the default `main` branch.
+Each snapshot reference has a uniquely identifiable name across all references of a table.
+A snapshot can have multiple references. The exact snapshot reference spec is documented at the [Spec](../spec/#snapshot-reference) page.
+Here we will provide some more explanations to the concepts in snapshot reference.
+
+### Reference Type
+
+There are clearly 2 types of snapshot reference, which are `branch` and `tag`. Their key differences are:
+
+- **New commit**: when a new snapshot is added as a child of a referenced snapshot, tag remains on the old snapshot, but branch reference moves to the child.
+
+- **Retention policy**: retention policy affects all the snapshots in a branch, but only a single tagged snapshot. (More details in the next section)
+
+### Retention Policy
+
+Iceberg offers a [snapshot expiration procedure](../spark-procedures/#expire_snapshots) to clean up snapshots that are not needed to free up storage space.
+Retention policy can be configured both globally and on snapshot reference to provide highly flexible customization to the expiration behavior.
+
+#### Global snapshot retention policy
+
+Global snapshot retention policy can be set through the following table properties:
+
+| Property                             | Default            | Description                                                   |
+| ------------------------------------ | ------------------ | ------------------------------------------------------------- |
+| history.expire.max-snapshot-age-ms   | 432000000 (5 days) | Default max age of snapshots to keep while expiring snapshots    |
+| history.expire.min-snapshots-to-keep | 1                  | Default min number of snapshots to keep while expiring snapshots |
+
+#### Snapshot reference retention policy
+
+Similarly, snapshot reference has the properties below to provider finer grain control:
+
+| Property                     | Type      | Description |
+|------------------------------|-----------|-------------|
+| **`min-snapshots-to-keep`**  | `int`     | For `branch` type only, the minimum number of snapshots to keep in a branch |
```

Comment:
How do we mention the branch name in the property?

Reply:
An update API will handle the actual update operation, which will allow specifying a branch name. It is not a part of this doc.